src/DynamoCoreWpf/ViewModels/Core/NoteViewModel.cs

```diff
@@ -305,11 +305,23 @@ private bool CanPinToNode(object parameters)
             // If there is no nodeModel or there is already a node pinned
             // Note cannot be pinned to any node
 
-            var nodeToPin = DynamoSelection.Instance.Selection
-                    .OfType<NodeModel>()
-                    .FirstOrDefault();
+            var nodeSelection = DynamoSelection.Instance.Selection
+                    .OfType<NodeModel>();
+
+            var noteSelection = DynamoSelection.Instance.Selection
+                    .OfType<NoteModel>();
+
+            if (nodeSelection == null || noteSelection == null || 
+                nodeSelection.Count() != 1 || noteSelection.Count() != 1)
+                return false;
+
+            var nodeToPin = nodeSelection.FirstOrDefault();
+
+            var nodeAlreadyPinned = WorkspaceViewModel.Notes
+                .Where(n => n.PinnedNode != null)
+                .Any(n => n.PinnedNode.NodeModel.GUID == nodeToPin.GUID);
 
-            if (nodeToPin == null || Model.PinnedNode != null)
+            if (nodeToPin == null || Model.PinnedNode != null || nodeAlreadyPinned)
```

Comment:
@martinstacey Can we add a unit test for this?

Reply:
Ok test added on this commit: 3a34af0